zipkin-lens/src/components/TracePage/SpanDetail/SpanDetail.jsx

```diff
@@ -24,51 +24,114 @@ import { detailedSpanPropTypes } from '../../../prop-types';
 const propTypes = {
   span: detailedSpanPropTypes.isRequired,
   minHeight: PropTypes.number.isRequired,
-  classes: PropTypes.shape({}).isRequired,
 };
 
-const style = theme => ({
+const useStyles = makeStyles(theme => ({
   root: {
     width: '100%',
     borderLeft: `1px solid ${theme.palette.grey[300]}`,
     backgroundColor: theme.palette.grey[100],
   },
-  names: {
-    borderBottom: `1px solid ${theme.palette.grey[300]}`,
+  serviceNameAndSpanName: {
+    paddingTop: theme.spacing(2),
+    paddingLeft: theme.spacing(2),
+    paddingBottom: theme.spacing(1.5),
+    paddingRight: theme.spacing(2),
   },
   serviceName: {
     textTransform: 'uppercase',
   },
   spanName: {
-    color: theme.palette.text.hint,
+    color: theme.palette.text.secondary,
   },
-});
+  annotationsAndTagsBox: {
+    paddingTop: theme.spacing(1),
+    paddingLeft: theme.spacing(2),
+    paddingRight: theme.spacing(2),
+    paddingBottom: theme.spacing(1.5),
+  },
+  annotationsAndTagsTitle: {
+    fontWeight: 'bold',
+    marginBottom: theme.spacing(0.5),
+  },
+  spanIds: {
+    display: 'flex',
+    justifyContent: 'flex-end',
+    paddingLeft: theme.spacing(2),
+    paddingRight: theme.spacing(2),
+    paddingTop: theme.spacing(0.1),
+    paddingBottom: theme.spacing(0.1),
+    fontSize: '0.8rem',
+    borderBottom: `1px solid ${theme.palette.grey[300]}`,
+  },
+  spanIdEntry: {
+    display: 'flex',
+  },
+  spanIdLabel: {
+    color: theme.palette.grey[700],
+    marginRight: theme.spacing(0.25),
+  },
+  spanIdValue: {
+    color: theme.palette.grey[800],
+    marginRight: theme.spacing(1),
+  },
+}));
 
-const SpanDetail = React.memo(({ span, minHeight, classes }) => (
-  <Box minHeight={minHeight} className={classes.root}>
-    <Box pt={2} pl={2} pb={1.5} pr={2} className={classes.names}>
-      <Typography variant="h5" className={classes.serviceName}>
-        {span.serviceName}
-      </Typography>
-      <Typography variant="h6" className={classes.spanName}>
-        {span.spanName}
-      </Typography>
+const SpanDetail = React.memo(({ span, minHeight }) => {
+  const classes = useStyles();
+
+  const spanIds = (
+    <Box className={classes.spanIds}>
+      {
+        [
+          { label: 'Span ID', value: span.spanId },
+          { label: 'Parent ID', value: span.parentId },
+        ].map((entry) => {
+          if (!entry.value) {
```

Comment:
What does it look like when there's no parent ID? Maybe rendering with `-` is better than skipping

Reply:
I see.